### PR TITLE
Add examples for persistence methods in Unity.

### PR DESCRIPTION
**Pull request by DominicFrei**
This PR contains the examples for the `persistence comparison` article that is to be released next.

Those are the typical ways I knew already or could find via research and chats with others. If there is a typical way Unity way to persist data please let me know.

Also, those examples make no sense in this repository. I just use it until we figured out how and where to put our examples. Will we create further repos per example? Will we create one example repo?

### CODE REVIEW

**Review by nirinchev — COMMENTED** [review]
Some general feedback that you may agree with or disregard:
1. I think you should decide between writing a comparison between different ways of saving structured data vs different ways of persisting data. `FileStreamExample` and `BinaryReaderWriterExample` are showcasing ways to persist data but don't really offer anything in terms of structure. It's easy to see how they would be inapplicable the moment you want to persist more than a single field. I think they are a little distracting and don't provide too much value - if you want to save some text to a file, it doesn't really matter whether you use `File.WriteAllLines` vs `File.WriteAllText`.
2. It might be a good idea to make the example self-contained by including all relevant classes in the same file. That way a reader looking to understand what is required to persist something with Realm doesn't have to jump between the example and the entity classes.
3. It might be a good idea to showcase persisting data immediately on user input rather than when the application quits. Since Unity is primarily used in mobile, there's no guarantee that `OnApplicationQuit` is going to be called and you don't want to lose user data.
4. It would be a good idea to make the example a little bit more complex - e.g. by having two counters - one for left clicks and a second one for right clicks. That way the differences between database approaches and json serialization will be more explicit.
5. Consider measuring the time to persist the data as that will certainly be a consideration for any Unity developer.

**Comment by DominicFrei** [comment]
> Some general feedback that you may agree with or disregard:
> 
> 1. I think you should decide between writing a comparison between different ways of saving structured data vs different ways of persisting data. `FileStreamExample` and `BinaryReaderWriterExample` are showcasing ways to persist data but don't really offer anything in terms of structure. It's easy to see how they would be inapplicable the moment you want to persist more than a single field. I think they are a little distracting and don't provide too much value - if you want to save some text to a file, it doesn't really matter whether you use `File.WriteAllLines` vs `File.WriteAllText`.
> 2. It might be a good idea to make the example self-contained by including all relevant classes in the same file. That way a reader looking to understand what is required to persist something with Realm doesn't have to jump between the example and the entity classes.
> 3. It might be a good idea to showcase persisting data immediately on user input rather than when the application quits. Since Unity is primarily used in mobile, there's no guarantee that `OnApplicationQuit` is going to be called and you don't want to lose user data.
> 4. It would be a good idea to make the example a little bit more complex - e.g. by having two counters - one for left clicks and a second one for right clicks. That way the differences between database approaches and json serialization will be more explicit.
> 5. Consider measuring the time to persist the data as that will certainly be a consideration for any Unity developer.

b088442 addresses (2)
2d7f309 addresses (3)

And (1) is a great question! When I initially started working on those examples they didn't have anything to do with Realm. My goal was to show every possible way to save data as a Unity developer. And I basically added everything that I knew plus everything that someone using Google would find with the goal of having all of those options in one place as a nice overview. And also showing an example for each of them.

The idea that built on that was then to use this and show that most solutions very quickly come to an end because they can only do simple stuff. That includes Unity's own `PlayerPrefs`.
You could technically save a JSON string using all of those solutions (which is even shown in one of those examples) but that leads to a whole bunch of other problems (manual migration just being one of them).

On the other side, the structured data approach would involve Json, Realm and SQL I assume?

I think both ways make sense but can't decide which route would be better. I'll also ping Shane, maybe he has some helpful insights on what resonates well with users.

I will address (4) + (5) as well.

File: PersistenceComparison/Assets/Scripts/RealmExample.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        {
+            // In case the database was empty, create a new `HitCountEntity`.
+            hitCountEntity = new HitCountEntity(1);
+            realm.Add(hitCountEntity);
+        }
+    }
+
+    private void OnApplicationQuit()
+    {
+
```
Comment (LaPeste) [inline PersistenceComparison/Assets/Scripts/RealmExample.cs:49]:
```suggestion

```

File: PersistenceComparison/Assets/Scripts/RealmExample.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            // In case the database was empty, create a new `HitCountEntity`.
+            hitCountEntity = new HitCountEntity(1);
+            realm.Add(hitCountEntity);
+        }
+    }
+
+    private void OnApplicationQuit()
+    {
+        
+
```
Comment (LaPeste) [inline PersistenceComparison/Assets/Scripts/RealmExample.cs:50]:
```suggestion
```

**Comment by DominicFrei** [comment]
Closed and moved to https://github.com/realm/unity-examples/pull/1, etc.

**Pull request closed without merging**